### PR TITLE
Check if when uri contains '/press-summary', the file name starts with 'Press Summary of'

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 1]
```diff
@@ -55,8 +55,12 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach with TableDrivenPro
   val sfnServer = new WireMockServer(9004)
   val testOutputBucket = "outputBucket"
   val inputBucket = "inputBucket"
-  val packageAvailable: TREInput = TREInput(TREInputParameters("status", "TEST-REFERENCE", inputBucket, "test.tar.gz"))
-  val event: SQSEvent = createEvent(packageAvailable.asJson.printWith(Printer.noSpaces))
+  private def packageAvailable(s3Key: String): TREInput = TREInput(
```
Comment (MancunianSam) [inline src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:58]: I think I told you this was the way to do it so sorry if I did but you don't need a second tar file. The metadata json for the tests comes from the `metadataJsonOpt` in the `stubAWSRequests` method so it doesn't need to be changed in the tar file. That means you can put these lines back the way they are on the main branch and then revert the changes from `event` to `event()`

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 2]
```diff
@@ -110,21 +114,21 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach with TableDrivenPro
 
   def stubAWSRequests(
       inputBucket: String,
-      fileDownloadBytes: Option[Array[Byte]] = None,
+      tarFileName: String = "test.tar.gz",
```
Comment (MancunianSam) [inline src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:117]: The changes in this method can be reverted as well because we don't need a different tar file.

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala [hunk 3]
```diff
@@ -299,14 +303,28 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach with TableDrivenPro
...
   }
 
+  "the lambda" should "error if the uri contains '/press-summary' but file name does not contain 'Press Summary of'" in {
+    val metadataJson: String =
+      s"""{"parameters":{"TDR": {"Document-Checksum-sha256": "abcde"},
+         |"TRE":{"reference":"$reference","payload":{"filename":"Test.docx"}},
+         |"PARSER":{"cite":"cite","uri":"https://example.com/id/cite/press-summary/3/","court":"test","date":"2023-07-26","name":"test"}}}""".stripMargin
+
+    val tarFileWithPressSummaryInUri = "PressSummaryOf-missing-from-name-test.tar.gz"
+    stubAWSRequests(inputBucket, tarFileName = tarFileWithPressSummaryInUri, metadataJsonOpt = Option(metadataJson))
```
Comment (MancunianSam) [inline src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:321]: This will need a change to do 
```scala
stubAWSRequests(inputBucket, metadataJsonOpt = Option(metadataJson))
```

File: src/main/scala/uk/gov/nationalarchives/Lambda.scala [hunk 4]
```diff
@@ -29,12 +29,22 @@ class Lambda extends RequestHandler[SQSEvent, Unit] {
...
         metadataFileInfo <- IO.fromOption(fileNameToFileInfo.get(s"$batchRef/TRE-$batchRef-metadata.json"))(
           new RuntimeException(s"Cannot find metadata for $batchRef")
         )
+
         treMetadata <- fileProcessor.readJsonFromPackage(metadataFileInfo.id)
-        parsedUri <- fileProcessor.parseUri(treMetadata.parameters.PARSER.uri)
+        uri = treMetadata.parameters.PARSER.uri
+        fileName = treMetadata.parameters.PARSER.name
+        _ <- {
+          val uriContainsPressSummary = uri.getOrElse("").contains("/press-summary")
```
Comment (MancunianSam) [inline src/main/scala/uk/gov/nationalarchives/Lambda.scala:40]: This is a style thing so feel free to ignore but what do you think about having:
```scala
          val uriContainsPressSummary = uri.exists(_.contains("/press-summary"))
          val fileNameDoesNotStartWithPressSummaryOf = !fileName.exists(_.startsWith("Press Summary of "))
```
It gets rid of the getOrElse("") at any rate. 

Also, what do you think about moving this block to a method in `FileProcessor` Most of the other lines in this for comprehension are single method calls and it can make it easier to read if you don't put too much logic in here.

**Review by MancunianSam — COMMENTED** [review]
Thanks for doing this. I forgot that the tar file contents weren't actually used. It has to be a tar file so the tar library can unpack it but once it does, it sends it as a PUT to the Wiremock S3 server which is lost so there isn't much need for two different ones.

File: src/test/scala/uk/gov/nationalarchives/LambdaTest.scala (hunk 1 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:58]: I've just seen the other PR that needs a different tar.gz file so it makes sense to leave this as you've done it. Ignore the comment on line 117 of LambdaTest as well.
You still don't need a new tar file for this PR though.

File: src/main/scala/uk/gov/nationalarchives/Lambda.scala (hunk 4 above)
Comment (techncl) [reply src/main/scala/uk/gov/nationalarchives/Lambda.scala:40]: 1. Ah, yes; that solution looks a lot cleaner 👍 
2. Regarding the 2nd solution, I wanted to have a call with you because I was thinking of lightly refactoring this `FileProcessor` file a little, as it's doing quite a bit which means that the `handleRequest` is hiding a lot of information making it harder to follow/debug. Regarding what you suggested, what if we created a `UriProcessor` Class, then I added a method with this press summary code in it and also moved the `parseUri` method from the `FileProcessor` Class to the `UriProcessor` Class (since parseUri is not doing any file processing per se)?

**Review by MancunianSam — APPROVED** [review]
Thanks for doing the refactoring. That's a lot better.
